src/components/PeopleFilters/FilterParts/NameFilter.tsx

```diff
@@ -0,0 +1,29 @@
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../../../utils';
+
+export const NameFilter = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+
+  return (
+    <div className="panel-block">
+      <p className="control has-icons-left">
+        <input
+          data-cy="NameFilter"
+          value={query}
+          onChange={event => setSearchParams(
+            getSearchWith(searchParams, {
+              query: event.target.value || null,
+            }),
+          )}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here